src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorProjectChangePublisher.cs

```diff
@@ -113,6 +113,12 @@ internal void EnqueuePublish(ProjectSnapshot projectSnapshot)
 
         internal void ProjectSnapshotManager_Changed(object sender, ProjectChangeEventArgs args)
         {
+            // Don't do any work if the solution is closing
+            if (args.SolutionIsClosing)
```

Comment:
Wouldn't that only matter in the case that you close the solution with the TagHelpers, then re-open it again in the same VS instance?

Reply:
If TagHelper resolution is fast enough it'd mean that we'd be rendering a partial `project.razor.json`. Aka, imagine you "close solution" and as part of that flow our project system starts getting its projects removed (to drain all state). Lets say we then publish a new project.razor.json with several projects removed in reaction. This would result in the next time the solution was opened reading a partial project.razor.json that doesn't actually have TagHelpers already serialized.